GCTC-NTGC/gc-digital-talent
# [Docs] Updates GitHub issue templates

## 👋 Introduction

This PR provides changes to several of the GitHub issue templates.

## 🕵️ Details

- [Updates new component template](https://github.com/GCTC-NTGC/gc-digital-talent/commit/92051db909176c8e5f828921fe0c31f42b99c03b) (layout more similar to other templates)
- [Removes design issue template](https://github.com/GCTC-NTGC/gc-digital-talent/commit/c7f3101d9e17b5955b4a79a96b93b78e046e23b5) (not used, design label doesn't even exist)
- [Removes new pattern issue template](https://github.com/GCTC-NTGC/gc-digital-talent/commit/1ac07621d32e44e1bb70d6bb864db34a6af66219) (not used, design label doesn't even exist)
- [Adds designer section](https://github.com/GCTC-NTGC/gc-digital-talent/commit/439e112b4c162440d09292ffcae6c7c3865a72e4) (provides attribution and a quick references for devs to see which designer worked on the original design)


## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Ensure templates have proper heading levels
2. Ensure templates are easy to use
